src/Sylius/Bundle/ApiBundle/StateProcessor/Admin/Channel/RemoveProcessor.php

```diff
@@ -22,30 +22,27 @@
 use Symfony\Component\HttpFoundation\Response;
 use Webmozart\Assert\Assert;
 
-/**
- * @implements ProcessorInterface<ChannelInterface>
- */
-final readonly class ChannelProcessor implements ProcessorInterface
+/** @implements ProcessorInterface<ChannelInterface> */
+final readonly class RemoveProcessor implements ProcessorInterface
 {
     public function __construct(
-        private ProcessorInterface $persistProcessor,
         private ProcessorInterface $removeProcessor,
         private ChannelDeletionCheckerInterface $channelDeletionChecker,
     ) {
     }
 
-    public function process($data, Operation $operation, array $uriVariables = [], array $context = [])
+    public function process($data, Operation $operation, array $uriVariables = [], array $context = []): void
     {
         Assert::isInstanceOf($data, ChannelInterface::class);
 
-        if (!$this->channelDeletionChecker->isDeletable($data) && $operation instanceof DeleteOperationInterface) {
-            throw new ChannelCannotBeRemoved('The channel cannot be deleted. At least one enabled channel is required.', Response::HTTP_UNPROCESSABLE_ENTITY);
+        if (!$operation instanceof DeleteOperationInterface) {
+            return;
         }
```

Comment:
This check is not consistent with other remove processors where are the asserts: `Assert::isInstanceOf($operation, DeleteOperationInterface::class);`